src/Microsoft.ComponentDetection.Contracts/ICommandLineInvocationService.cs

```diff
@@ -53,6 +55,26 @@ public interface ICommandLineInvocationService
     /// <param name="parameters">The parameters that should be passed to the command. The parameters will be space-joined.</param>
     /// <returns>Awaitable task with the result of executing the command, including exit code.</returns>
     Task<CommandLineExecutionResult> ExecuteCommandAsync(string command, IEnumerable<string> additionalCandidateCommands = null, params string[] parameters);
+
+    /// <summary>
+    /// Executes a command line command. If the command has not been located yet, CanCommandBeLocated will be invoked without the submitted parameters.
+    /// </summary>
+    /// <param name="command">The command name to execute. Environment variables like PATH on windows will also be considered if the command is not an absolute path. </param>
+    /// <param name="additionalCandidateCommands">Other commands that could satisfy the need for the first command. Assumption is that they all share similar calling patterns.</param>
+    /// <param name="workingDirectory">The directory under which to run the command.</param>
+    /// <param name="parameters">The parameters that should be passed to the command. The parameters will be space-joined.</param>
+    /// <returns>Awaitable task with the result of executing the command, including exit code.</returns>
+    Task<CommandLineExecutionResult> ExecuteCommandAsync(string command, IEnumerable<string> additionalCandidateCommands = null, DirectoryInfo workingDirectory = null, params string[] parameters);
```

Comment:
It looks like you updated almost all of the `ExecuteCommandAsync` calls to include a CancellationToken already - do we want to expose this without one? Requiring it for future detectors would be a best practice

Reply:
So I left these around because it looks like its part of the exposed library api. We are using it in CD-internal, and removing this would be a breaking change. Do you know if there are other consumers who we have to be cognizant of, SBOM maybe?

Reply:
SBOM interacts directly with the ScanCommand class - they don't attempt to interface directly with the services.

Reply:
Going to mark these as obselete for now